src/requests/getForecast.js

```diff
@@ -0,0 +1,35 @@
+import axios from "axios";
+
+const getForecast = async (
+  setSelectedDate,
+  setForecasts,
+  setLocation,
+  searchText,
+  setErrorMessage,
+) => {
+  let response;
+  try {
+    if (!searchText) {
+      response = await axios.get(`https://cmd-shift-weather-app.onrender.com/forecast?city=London`);
+    } else {
+      setErrorMessage("");
+      response = await axios.get(`https://cmd-shift-weather-app.onrender.com/forecast?city=${searchText}`);
+    }
+  } catch (error) {
+    if (error.response.status === 404) {
+      // eslint-disable-next-line no-useless-escape
+      setErrorMessage(`Sorry, \" ${searchText} \"  is not a valid town or city in the UK. Try again.`);
+      // eslint-disable-next-line no-console
+      console.error("Location is not valid", error);
+    } else {
+      setErrorMessage("Server error, try again later!");
+      // eslint-disable-next-line no-console
+      console.error("Server Error", error);
+    }
+  }
+  setSelectedDate(response.data.forecasts[0].date);
+  setForecasts(response.data.forecasts);
+  setLocation(response.data.location);
```

Comment:
So there are a few things to look at here 
- You dont need all the eslint exceptions as you can write the output you want without having to do things in a way that necessitates it
- The console logs arent needed as you are displaying something on the screen in the event of failure 
- Theres a bug which shows when you enter a non valid town into the input: it crashes the app as you are trying to pass values from a response which doesnt exit into the setters:
```
  try {
    if (!searchText) {
      response = await axios.get(
        `https://cmd-shift-weather-app.onrender.com/forecast?city=London`
      );
    } else {
      setErrorMessage("");
      response = await axios.get(
        `https://cmd-shift-weather-app.onrender.com/forecast?city=${searchText}`
      );
    }
    setSelectedDate(response.data.forecasts[0].date);
    setForecasts(response.data.forecasts);
    setLocation(response.data.location);
  } catch (error) {
    if (error.response.status === 404) {
      setErrorMessage(
        `Sorry, " ${searchText} "  is not a valid town or city in the UK. Try again.`
      );
    } else {
      setErrorMessage("Server error, try again later!");
    }
  }
```